uspace/lib/cpp/include/__bits/locale/num_get.hpp

```diff
@@ -300,18 +300,20 @@ namespace std
                 auto size = fill_buffer_integral_(in, end, base);
                 if (size > 0)
                 {
-                    int ret{};
+                    int olderrno{errno};
+                    errno = EOK;
+
                     if constexpr (is_signed<BaseType>::value)
-                        ret = std::hel::str_int64_t(base.buffer_, nullptr, num_base, false, &res);
+                        res = ::strtoll(base.buffer_, nullptr, num_base);
                     else
-                        ret = std::hel::str_uint64_t(base.buffer_, nullptr, num_base, false, &res);
+                        res = ::strtoull(base.buffer_, nullptr, num_base);
 
-                    if (ret != EOK)
-                    {
+                    if (errno != EOK)
                         err |= ios_base::failbit;
-                        v = 0;
```

Comment:
The zero assignment here is required by $22.4.2.1.2 (n4296 [1] bottom of page 717, point (3.4)). This is because if we use something like:

    int x = 3;
    std::cin >> x;

And we input something that isn't convertible, the value of `x` should be `0`, not `3`.

Reply:
[1] http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf

Reply:
`x` is assigned later. The original behavior wouldn't be consistent with the standard, see behavior of `strtoll()`:
>If the value read is out of the range of representable values by a long long int, the function returns LLONG_MAX or LLONG_MIN (defined in <climits>), and errno is set to ERANGE.

Reply:
Actually, now that I'm looking at it, the C function doesn't guarantee `errno` being set at all when there are no digits in the input. This needs an extra check.

Reply:
So I dug a little deeper and discovered a bunch of problems in implementation of both the standard and nonstandard conversion functions. I suppose it's as good a time as any to finally start making pcut tests...